### PR TITLE
Fix vulnerability in release-1.23

### PR DESCRIPTION
**Pull request by sawsa307**
Fix vulnerability in release-1.23.
* Bump go version to 1.20.10.
* Bump golang.org/x/net version to v0.13.0.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -27,6 +27,8 @@ require (
 	k8s.io/utils v0.0.0-20221107191617-1a15be271d1d
 )
 
+require k8s.io/cloud-provider-gcp/crd v0.0.0-20230411231432-6ab330834e4e
```
Comment (swetharepakula) [inline go.mod:30]: why did this move out here?

**Review by swetharepakula — COMMENTED** [review]
One question about the go.mod but otherwise LGTM

/ok-to-test

File: go.mod (hunk 1 above)
Comment (sawsa307) [reply go.mod:30]: That's some automatic formatting did by `go mod tidy`.

Comment (swetharepakula) [reply go.mod:30]: thanks for clarifying!

**Pull request merged**